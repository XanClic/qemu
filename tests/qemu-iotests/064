#!/bin/bash
#
# Test for additional information emitted by qemu-img info on qcow2
# images
#
# Copyright (C) 2013 Red Hat, Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

# creator
owner=mreitz@redhat.com

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!

_cleanup()
{
	_cleanup_test_img
}
trap "_cleanup; exit \$status" 0 1 2 3 15

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

# This tests qocw2-specific low-level functionality
_supported_fmt qcow2
_supported_proto generic
_supported_os Linux

IMG_SIZE=64M

echo
echo "=== Testing qcow2 image with -o compat=0.10 ==="
echo
IMGOPTS="compat=0.10" _make_test_img $IMG_SIZE
# don't use _img_info, since that function will filter out the
# additional information we're about to test for
$QEMU_IMG info "$TEST_IMG" | grep "Format specific information:" -A 42

echo
echo "=== Testing qcow2 image with -o compat=1.1,lazy_refcounts=off ==="
echo
IMGOPTS="compat=1.1,lazy_refcounts=off" _make_test_img $IMG_SIZE
$QEMU_IMG info "$TEST_IMG" | grep "Format specific information:" -A 42

echo
echo "=== Testing qcow2 image with -o compat=1.1,lazy_refcounts=on ==="
echo
IMGOPTS="compat=1.1,lazy_refcounts=on" _make_test_img $IMG_SIZE
$QEMU_IMG info "$TEST_IMG" | grep "Format specific information:" -A 42

# success, all done
echo "*** done"
rm -f $seq.full
status=0
