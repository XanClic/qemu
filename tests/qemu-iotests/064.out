QA output created by 064

 === Testing image size bounds ===

Formatting 'TEST_DIR/t.IMGFMT', fmt=IMGFMT size=67108864
Test 1: Maximum size (1024 TB):
image: TEST_DIR/t.IMGFMT
file format: IMGFMT
virtual size: 1024T (1125899905794048 bytes)
cluster_size: 1048576

Test 2: Size too large (1024TB + 1)
qemu-img: Could not open 'TEST_DIR/t.IMGFMT': Invalid argument

Test 3: Size valid (64M), but Blocks In Image too small (63)
qemu-img: Could not open 'TEST_DIR/t.IMGFMT': Operation not supported

Test 4: Size valid (64M), but Blocks In Image exceeds max allowed
qemu-img: Could not open 'TEST_DIR/t.IMGFMT': Operation not supported

Test 5: Valid Image: 64MB, Blocks In Image 64, Block Size 1MB
image: TEST_DIR/t.IMGFMT
file format: IMGFMT
virtual size: 64M (67108864 bytes)
cluster_size: 1048576

Test 6: Block Size != 1MB; too small test (1MB - 1)
qemu-img: Could not open 'TEST_DIR/t.IMGFMT': Operation not supported

Test 7: Block Size != 1MB; too large test (1MB + 1)
qemu-img: Could not open 'TEST_DIR/t.IMGFMT': Operation not supported

*** done
